creator-node/src/routes/files.js

```diff
@@ -157,7 +158,9 @@ const getCID = async (req, res) => {
 
   try {
     // Add a rehydration task to the queue to be processed in the background
-    RehydrateIpfsQueue.addRehydrateIpfsFromFsIfNecessaryTask(CID, storagePath, { logContext: req.logContext })
+    if (enableRehydrate) {
```

Comment:
a much cleaner solution would be to put this if check inside the `addRehydrateIpfsFromFsIfNecessaryTask()` function. you already changed the name to indicate ifNecessary, the caller should not need to worry about this logic, it should just call it and the function internally either no-ops or actually proceeds. would also be much less code

Reply:
will update!